x-pack/osquerybeat/internal/osqd/osqueryd.go

```diff
@@ -182,6 +184,13 @@ func (q *OSQueryD) Run(ctx context.Context, flags Flags) error {
 		return err
 	}
 
+	// Assign osqueryd process to the JobObject on windows
+	// in order to assure no orphan process is left behind
+	// after osquerybeat process is killed.
+	if err := proc.JobObject.Assign(cmd.Process); err != nil {
+		q.log.Errorf("osqueryd process failed job assign: %v", err)
+	}
+
```

Comment:
Something to consider, we are working on moving Agents outside of the beats repository this adds a dependency between the two. We should consider either duplicating the code.

Reply:
I thought we are moving libbeat in it's own repository, and the agent and the beats will have their own separate repositories. In this case both the agent code and the osquerybeat code will import dependency from libbeat as it is done with many other shared libbeat pieces.